src/xcode/ENA/ENA/Source/Scenes/HealthCertificate/Cells/HealthCertificateQRCodeCellViewModel.swift

```diff
@@ -0,0 +1,46 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+
+struct HealthCertificateQRCodeCellViewModel {
+
+	// MARK: - Init
+
+	init(
+		healthCertificate: HealthCertificateData
+	) {
+		var dateOfVaccination: String = ""
+		if let vaccinationDate = healthCertificate.dateOfVaccination {
+			dateOfVaccination = DateFormatter.localizedString(from: vaccinationDate, dateStyle: .medium, timeStyle: .none)
+		}
+		let expirationDate = DateFormatter.localizedString(from: healthCertificate.expirationDate, dateStyle: .medium, timeStyle: .none)
+		self.validity = String(
+			format: AppStrings.HealthCertificate.Details.validity,
+			dateOfVaccination, expirationDate
+		)
+		self.certificate = String(
+			format: AppStrings.HealthCertificate.Details.certificateCount,
+			healthCertificate.doseNumber, healthCertificate.totalSeriesOfDoses
+		)
+
+		let qrCodeSize = UIScreen.main.bounds.width - 60
```

Comment:
Would be nice to work with a percentage of the width, so it does not steal to much from a small device and not to less from a huge device. Nice to have not blocking.

Reply:
I think the goal here is actually to have the QR code as big as possible, as it contains a lot of information, so we use the same margins for every device.